src/mapper/pkg/resolvers/helpers.go

```diff
@@ -73,3 +111,70 @@ func (i *intentsHolder) GetIntentsPerService(namespaces []string) map[model.Otte
 	}
 	return result
 }
+
+func (i *intentsHolder) getConfigMapNamespace() (string, error) {
+	namespace := viper.GetString(config.NamespaceKey)
+	if namespace != "" {
+		return namespace, nil
+	}
+	namespace, err := kubeutils.GetCurrentNamespace()
+	if err != nil {
+		return "", fmt.Errorf("could not deduce the store's configmap namespace: %w", err)
+	}
+	return namespace, nil
+}
+
+func (i *intentsHolder) WriteStore(ctx context.Context) error {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+	jsonBytes, err := json.Marshal(i.store)
```

Comment:
should this be `i.store.MarshalJSON()`?

Reply:
That's what json.Marshal() does :) it calls the MarshalJSON() if an object if it has one.